### ISSUE TITLE
Mayavi installation fails: 'NoneType' object has no attribute 'find'

### ISSUE THREAD

**Issue by hohlraum** [enthought/mayavi#159]
Hi,

I have working copies of the dependencies (vtk, traits, etc...), now I'm trying to install Mayavi. The process fails when using both pip or setup.py. The tail of the traceback is included below. I can include the whole thing (it's very long) if it would help. A debug postmortem shows that it fails on the method ```EventIds```

I'm running this under Mac OSX 10.9.5, with Python 2.7.8

```      
...
File "tvtk/wrapper_gen.py", line 734, in _gen_other_methods
  self._write_tvtk_method(out, vtk_meth)
File "tvtk/wrapper_gen.py", line 1035, in _write_tvtk_method
  sig = self.parser.get_method_signature(vtk_meth)
File "tvtk/vtk_parser.py", line 304, in get_method_signature
  doc = doc[:doc.find('\n\n')]
AttributeError: 'NoneType' object has no attribute 'find'
```

**Comment by ariell** [comment]
I am having the same issues. Any resolution on this?

**Comment by myfavecolorisgreen** [comment]
I'm having the same problem too. I think mayavi installed okay the first time I tried pip install mayavi, but I hadn't correctly installed vtk at that point, so import mayavi did not work. After getting vtk installed correctly, I ran pip uninstall mayavi, thinking that I should try to reinstall it. Now, I'm not able to install mayavi, and get the same error as above. 

I have tried pip uninstall-ing all of mayavi's dependencies and then running pip install mayavi again, but this didn't work either.

**Comment by ElliotKaplan** [comment]
I had the same problem (Python 2.7.8, Ubuntu 14.04). The workaround that I settled on was to catch the error in  _write_tvtk_method and ignore it

        if not sig:
            try:
                sig = self.parser.get_method_signature(vtk_meth)
            except:
                print 'err159\t', vtk_meth
                return

The build trace can be seen here http://pastebin.ca/2985442
This resulted in a mostly usable version of Mayavi2---vector and scalar cut planes are not working properly, which may be a function of something else---though I'm not sure exactly what's being swept under a rug here. The print statement gets triggered for a bunch of <type 'SomeType'> objects.

**Comment by ckalas** [comment]
@ElliotKaplan thanks, this allowed me to finally build it too.

**Comment by jjstickel** [comment]
A patch for the workaround described by ElliotKaplan can be found here:
https://trac.macports.org/attachment/ticket/47762/mayavi_wrapper.patch

**Comment by itziakos** [comment]
What is the vtk version that results in this error?

**Comment by jenshnielsen** [comment]
I have seen this with VTK 6.2 (installed with brew on OSX) I don't have the machine with me at this stage so I cannot investigate further.

**Comment by dmsurti** [comment]
@jenshnielsen @itziakos Elliot's is a fix that won't really work as the VTK parser in tvtk package needs to be updated to handle API changes in VTK 6.2 api. 

The build trace by @ElliotKaplan is very useful though to investigate further.  http://pastebin.ca/2985442

**Comment by dmsurti** [comment]
Fixed in #191
